### ISSUE TITLE
[Feature Request] List of Links in Side Pane

### ISSUE THREAD

**Issue by ryadaj** [TfTHacker/obsidian42-strange-new-worlds#104]
It could be interesting to present an alternate view, instead of the counters which appear next to links in the content of a file.

Instead, the plugin could have a menu in the side pane which presents the "Outgoing links from the current file" in a list. It would be cool if the links in this list could then be unfolded by clicking an arrow or something. Upon doing so, a list of files which contain that link appears AND/OR the context in which that link appears is presented.

Thoughts?

**Comment by TfTHacker** [comment]
This is very interesting. Could you mock up a screenshot to illustrate the idea?

**Comment by ryadaj** [comment]
Well, as things are currently with SNW, you have an active note, with counters next to each link indicating how many notes there are in the vault which contain this link (you know this because you designed the plugin xD) 

My suggestion is to replace the counters with an "enhanced outgoing links" pane. On the right of the screenshot below is the core outgoing links pane. I propose adding a toggle next to each link in the outgoing links pane, which when clicked will unfold and present the same view that appears when clicking/hovering over one of SNW's counters: a list of files which contain the link in question and the context in which that link occurs.

Does this make any sense? 🫠

<img width="1800" alt="Screenshot 2023-07-22 at 11 41 30" src="https://github.com/TfTHacker/obsidian42-strange-new-worlds/assets/82765301/004c6101-a3e9-414b-9dd0-c19303e96a92">

**Comment by ryadaj** [comment]
Did that make any sense? 😂 Basically the benefit would be the option to see the outgoing links from the active note presented in a list in the sidebar, and the option to click each of these to fold/unfold into a list of notes which also link to, e.g., "Fear", "Bravery", etc. in the above example.

**Comment by TfTHacker** [comment]
I do like the idea. This would basically serve as a replacement for the built-in outgoing links pane?

**Comment by ryadaj** [comment]
Yes! I picture it being called "shared links" haha, but yes, it could be thought of as an enhanced outgoing links pane

**Comment by ryadaj** [comment]
Ooooo. Just saw [this message](https://discord.com/channels/686053708261228577/744933215063638183/1162850144794988735) on Discord. This is exactly what I was thinking here, but for links and living in one of the side panels!
